distribution/docs/pom.xml

```diff
@@ -119,6 +122,42 @@
     </profiles>
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.codehaus.gmaven</groupId>
+                <artifactId>groovy-maven-plugin</artifactId>
+                <version>2.1</version>
+                <inherited>false</inherited>
+                <executions>
+                    <execution>
+                        <id>aggregate-deps</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>execute</goal>
+                        </goals>
+                        <configuration>
+                            <source>
+                                def lines = []
+                                new File('${highest-basedir}').traverse(type: groovy.io.FileType.FILES, nameFilter: ~/maven-deps.txt/) { file ->
+                                    for (line in file.readLines())
+                                        lines.add(line.trim())
+                                }
+                                def output = ""
+                                Outer:
+                                for (line in lines.unique().sort()) {
```

Comment:
Instead of calling unique on lines, could we instead store them in a Set?

We could also pipeline some of this instead (as well as getting rid of the labels), with something like:
```groovy
Set lines = []
Set ignoredLines = ${dependency-ignored-lines} as Set
List ignoredPackages = ${dependency-ignored-packages}
new File('${highest-basedir}').traverse(type: groovy.io.FileType.FILES, nameFilter: ~/maven-deps.txt/) { file ->
    for (line in file.readLines())
        String depLine = line.trim()
        if (!(depLine in ignoredLines) && !ignoredPackages.any {line.startsWith(it)}) {
            lines.add(depLine)
        }
}
String output = lines
    .sort()
    .collect {"* ${line}"}
    .join(System.lineSeparator())
```